lib/core/api/get-file.js

```diff
@@ -24,43 +27,98 @@ module.exports = function(self) {
     throw e;
   };
 
+  function createFileToken(file, cb) {
+    file._client.createToken(file._bucketId, 'PULL', function(e, body) {
```

Comment:
My thinking was that we could push the dependency on `self` down into the constructor to decouple the two structures.